spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/couchbase/CouchbaseConfigurerAdapterConfiguration.java

```diff
@@ -37,16 +41,21 @@
 
 	private final CouchbaseConfiguration configuration;
 
-	CouchbaseConfigurerAdapterConfiguration(CouchbaseConfiguration configuration) {
+	private final CouchbaseEnvironment couchbaseEnvironment;
+
+	private final Cluster cluster;
+
+	CouchbaseConfigurerAdapterConfiguration(CouchbaseConfiguration configuration,
+			CouchbaseEnvironment couchbaseEnvironment, Cluster cluster) {
 		this.configuration = configuration;
+		this.couchbaseEnvironment = couchbaseEnvironment;
+		this.cluster = cluster;
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
 	public CouchbaseConfigurer springBootCouchbaseConfigurer() throws Exception {
-		return new SpringBootCouchbaseConfigurer(
-				this.configuration.couchbaseEnvironment(),
-				this.configuration.couchbaseCluster(),
+		return new SpringBootCouchbaseConfigurer(this.couchbaseEnvironment, this.cluster,
```

Comment:
I'm worried that this may introduce some subtle bugs. Previously, the four components passed into `SpringBootCouchbaseConfigurer` all came came from the same source. That's no longer guaranteed to be the case. In particular using the cluster from any source and the cluster info from `CouchbaseConfiguration` doesn't feel quite right. I'm not sure what the answer is yet.